src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2120,6 +2140,21 @@ def set_up_kube_proxy_config(self, mc: ManagedCluster) -> ManagedCluster:
         mc.network_profile.kube_proxy_config = self.context.get_kube_proxy_config()
         return mc
 
+    def set_up_auto_upgrade_profile(self, mc: ManagedCluster) -> ManagedCluster:
+        """Set up auto upgrade profile for the ManagedCluster object.
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        mc = super().set_up_auto_upgrade_profile(mc)
```

Comment:
@FumingZhang, curious how we should handle decorator methods that exist both in the core cli, and the preview. Is this the correct patterning? I didn't spot another example, but in this case the upgrade channel is GA and supported in the core CLI, while we're adding a new property node_os_upgrade_channel that is only supported in the preview.

Reply:
Yes, this is the correct pattern (reuse the code and add some extensions).